cylc/flow/cfgspec/globalcfg.py

```diff
@@ -1493,9 +1493,11 @@ def load(self):
                     LOG.error(f'bad {conf_type} {fname}')
                     raise
 
+        # Expand platforms needs to be performed first because it
+        # manipulates the sparse config.
+        self._expand_platforms()
         self._set_default_editors()
```

Comment:
I think the fact that the dense config is expanded in a method called `_set_default_editors()` is a bit troublesome. I am addressing this as part of #4815